qcodes/instrument_drivers/Keysight/N9030B.py

```diff
@@ -0,0 +1,601 @@
+                             stop: float,
+                             npts: int) -> None:
+        """
+        Sets up the Swept SA measurement sweep for Spectrum Analyzer Mode.
+        """
+        self.root_instrument.mode("SA")
+        if "SAN" in self.root_instrument._available_meas():
+            self.root_instrument.measurement("SAN")
+        else:
+            raise RuntimeError("Swept SA measurement is not available on your "
+                               "Keysight N9030B instrument with Spectrum "
+                               "Analyzer mode.")
+        self.start(start)
+        self.stop(stop)
+        self.npts(npts)
+
+    def autotune(self) -> None:
+        """
+        Autotunes frequency
+        """
+        self.write(":SENS:FREQuency:TUNE:IMMediate")
+
+
+class PhaseNoiseMode(InstrumentChannel):
+    """
+    Phase Noise Mode for Keysight N9030B instrument.
+    """
+
+    def __init__(self, parent: "N9030B", name: str, *arg: Any, **kwargs: Any):
+        super().__init__(parent, name, *arg, **kwargs)
+
+        self._min_freq = 1
+        self._valid_max_freq: Dict[str, float] = {"503": 3699999995,
+                                                  "508": 8499999995,
+                                                  "513": 13799999995,
+                                                  "526": 26999999995,
+                                                  "544": 44499999995}
+        opt: str
+        for hw_opt_for_max_freq in self._valid_max_freq.keys():
+            if hw_opt_for_max_freq in self.root_instrument._options():
+                opt = hw_opt_for_max_freq
+        self._max_freq = self._valid_max_freq[opt]
+
+        self.add_parameter(
+            name="npts",
+            get_cmd=":SENSe:LPLot:SWEep:POINts?",
+            set_cmd=":SENSe:LPLot:SWEep:POINts {}",
+            get_parser=int,
+            vals=Ints(601, 20001),
+            docstring="Number of points for the sweep"
+        )
+
+        self.add_parameter(
+            name="start_offset",
+            get_cmd=":SENSe:LPLot:FREQuency:OFFSet:STARt?",
+            set_cmd=self._set_start_offset,
+            get_parser=float,
+            vals=Numbers(self._min_freq, self._max_freq - 10),
+            docstring="start frequency offset for the plot"
+        )
+
+        self.add_parameter(
+            name="stop_offset",
+            get_cmd=":SENSe:LPLot:FREQuency:OFFSet:STOP?",
+            set_cmd=self._set_stop_offset,
+            get_parser=float,
+            vals=Numbers(self._min_freq + 99, self._max_freq),
+            docstring="stop frequency offset for the plot"
+        )
+
+        self.add_parameter(
+            name="signal_tracking_enabled",
+            get_cmd=":SENSe:FREQuency:CARRier:TRACk?",
+            set_cmd=":SENSe:FREQuency:CARRier:TRACk {}",
+            val_mapping=create_on_off_val_mapping(on_val="ON", off_val="OFF"),
+            docstring="Gets/Sets signal tracking"
```

Comment:
Could this docstring be improved to explain what signal tracking is?

Reply:
Added more information to the doc string about signal tracking